docs/security.html

```diff
@@ -257,93 +321,191 @@
-            <pre>
-    -Djava.security.krb5.conf=/etc/kafka/krb5.conf
+              </li>
+              <li>Pass the JAAS and optionally krb5 file locations as JVM parameters to each client JVM (see <a href="https://docs.oracle.com/javase/8/docs/technotes/guides/security/jgss/tutorials/KerberosReq.html">here</a> for more details): 
+              <pre>    -Djava.security.krb5.conf=/etc/kafka/krb5.conf
     -Djava.security.auth.login.config=/etc/kafka/kafka_client_jaas.conf</pre></li>
-            <li>Make sure the keytabs configured in the kafka_client_jaas.conf are readable by the operating system user who is starting kafka client.</li>
-            <li>Configure the following properties in producer.properties or consumer.properties:
-                <pre>
-    security.protocol=SASL_PLAINTEXT (or SASL_SSL)
-    sasl.kerberos.service.name=kafka</pre>
-            </li>
-        </ol></li>
-
-    <li><h4><a id="security_rolling_upgrade" href="#security_rolling_upgrade">Incorporating Security Features in a Running Cluster</a></h4>
-        You can secure a running cluster via one or more of the supported protocols discussed previously. This is done in phases:
-        <p></p>
+              <li>Make sure the keytabs configured in the kafka_client_jaas.conf are readable by the operating system user who is starting kafka client.</li>
+              <li>Configure the following properties in producer.properties or consumer.properties: 
+              <pre>    security.protocol=SASL_PLAINTEXT (or SASL_SSL)
+    sasl.mechanism=GSSAPI
+    sasl.kerberos.service.name=kafka</pre></li>
+          </ol>
+      </li>
+    </ol>
+  </li>
+      
+  <li><h4><a id="security_sasl_plain" href="#security_sasl_plain">Authentication using SASL/PLAIN</a></h4>
+    <p>SASL/PLAIN is a simple username/password authentication mechanism that is typically used with TLS for encryption to implement secure authentication.
+       Kafka supports a default implementation for SASL/PLAIN which can be extended for production use as described <a href="#security_sasl_plain_production">here</a>.</p>
+       The username is used as the authenticated <code>Principal</code> for configuration of ACLs etc.
+    <ol>
+      <li><h5><a id="security_sasl_plain_brokerconfig" href="#security_sasl_plain_brokerconfig">Configuring Kafka Brokers</a></h5>
+        <ol>
+          <li>Add a suitably modified JAAS file similar to the one below to each Kafka broker's config directory, let's call it kafka_server_jaas.conf for this example:
+            <pre>
+    KafkaServer {
+        org.apache.kafka.common.security.plain.PlainLoginModule required
+        username="admin"
+        password="admin-secret"
+        user_admin="admin-secret"
+        user_alice="alice-secret";
+    };</pre></li>
+          This configuration defines two users (<i>admin</i> and <i>alice</i>). <i>admin</i> is used for inter-broker communication.
```

Comment:
Yes, second on Magnus question. We should make it clear that username/password is used for broker to initiate a connection to other brokers and user_userName is used for broker to validate all client connections (including those from other brokers).